tests/int/test_headers.py

```diff
@@ -0,0 +1,174 @@
+
+        #Teardown
+        print('Cleanup product')
+        await product.destroy_product()
+
+
+    @pytest.fixture
+    async def app(self, asid = '1234567890'):
+         # Setup
+        app = ApigeeApiDeveloperApps()
+        await app.create_new_app()
+
+        # Set default JWT Testing resource url and ASID
+        await app.set_custom_attributes({
+                'asid': asid,
+                'jwks-resource-url': 'https://raw.githubusercontent.com/NHSDigital/'
+                                     f'identity-service-jwks/main/jwks/{ENVIRONMENT}/'
+                                     '9baed6f4-1361-4a8e-8531-1f8426e3aba8.json'
+        })
+        yield app
+
+        #Teardown
+        print('Cleanup app')
+        await app.destroy_app()
+
+
+    async def get_access_code(self, appToUse, user_id):
+        # Create an OAuthHelper instance with the new app
+        oauth = OauthHelper(appToUse.client_id, appToUse.client_secret, appToUse.callback_url)
+
+        with open(ID_TOKEN_TESTING_PRIVATE_KEY_PATH, "r") as f:
+            contents = f.read()
+
+        claims = {
+                'at_hash': 'tf_-lqpq36lwO7WmSBIJ6Q',
+                'sub': user_id,
+                'auditTrackingId': '91f694e6-3749-42fd-90b0-c3134b0d98f6-1546391',
+                'amr': ['N3_SMARTCARD'],
+                'iss': 'https://am.nhsint.auth-ptl.cis2.spineservices.nhs.uk:443/'
+                       'openam/oauth2/realms/root/realms/NHSIdentity/realms/Healthcare',
+                'tokenName': 'id_token',
+                'aud': '969567331415.apps.national',
+                'c_hash': 'bc7zzGkClC3MEiFQ3YhPKg',
+                'acr': 'AAL3_ANY',
+                'org.forgerock.openidconnect.ops': '-I45NjmMDdMa-aNF2sr9hC7qEGQ',
+                's_hash': 'LPJNul-wow4m6Dsqxbning',
+                'azp': '969567331415.apps.national',
+                'auth_time': 1610559802,
+                'realm': '/NHSIdentity/Healthcare',
+                'exp': int(time()) + 6000,
+                'tokenType': 'JWTToken',
+                'iat': int(time()) - 100
+        }
+
+        print('Claims:')
+        print(claims)
+
+        client_assertion_jwt = oauth.create_jwt(kid='test-1')
+        id_token_jwt = oauth.create_id_token_jwt(algorithm="RS512", kid="identity-service-tests-1", claims=claims, signing_key=contents)
+        # When
+        resp = await oauth.get_token_response(
+            grant_type="token_exchange",
+            _jwt=client_assertion_jwt,
+            id_token_jwt=id_token_jwt
+        )
+        print('JWT Token response:')
+        print(resp)
+
+        return resp['body']['access_token']
+
+
+    @pytest.mark.asyncio
+    async def test_headers(self, product, app):
+        # Assign the new product to the app
+        await app.add_api_product([product.name])
+
+        user_id = '0987654321123'
+        access_code = await self.get_access_code(app, user_id)
+
+        client_request_headers = {
+            HEADER_ECHO : '',
```

Comment:
What does setting the "echo" header do?

Reply:
It now enables the echo target. The path is no longer used.

Reply:
I see that now thanks.

Reply:
Maybe worth a comment to explain?